nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerASMDump.java

```diff
@@ -70,6 +70,14 @@
             mv.visitMethodInsn(INVOKESTATIC, "edu/illinois/nondex/shuffling/ControlNondeterminism", "shuffle", "(Ljava/util/List;)Ljava/util/List;", false);
             mv.visitVarInsn(ASTORE, 3);
             mv.visitVarInsn(ALOAD, 0);
+            mv.visitFieldInsn(GETFIELD, baseName + "$HashIterator$HashIteratorShuffler", "hashIter", "L" + baseName + "$HashIterator;");
+            mv.visitInsn(ACONST_NULL);
+            if("java/util/WeakHashMap".equals(baseName)){
+                mv.visitFieldInsn(PUTFIELD, baseName + "$HashIterator", "lastReturned", "L" + baseName + "$" + type + ";");
```

Comment:
I see you added the sources. But they are not the modified sources to refer to the shuffler. 
1) Can you add a comment where the shuffler would be used?
2) As I said earlier, remove this IF and pass the name of the field as argument in the Instrumenter, i.e., lastReturned.